### ISSUE TITLE
Hide or implement bionic slot restrictions for 0.D

### ISSUE THREAD

**Issue by Coolthulhu** [CleverRaven/Cataclysm-DDA#20305]
Since 0.D is supposed to be stable, it shouldn't have the halfway-implemented bionic slot system shown to players.
Implementing it could be a bit risky (not enough time to test properly), so a switch of some sort to simply prevent displaying them could be a better way to solve this.

**Comment by DangerNoodle** [comment]
Hiding the system if it cannot be implemented in time would likely be the best solution. Every new feature, especially one that may adversely alter gameplay balance and cause issue discussion to get bogged down by potential player complaints, would hinder both fixing it and hinder preparations for the next stable release.

If anything, the system should not have been implemented in the first place if it was not functional as-is. Same reason for my primary objection to the nutrition system, as allegedly the actual consequences were not coded to have any effect when first added.

**Comment by WizardOfOoo** [comment]
It is confusing to players, I see questions on Reddit about it often.

**Comment by CoroNaut** [comment]
The major concerns would be: Does it have bugs" Will the community take it well? Is it balanced enough? We currently don't have an answer for all 3, so I suggest hiding the feature.

**Comment by DangerNoodle** [comment]
Agreed.

**Comment by keyspace** [comment]
> it shouldn't have the halfway-implemented bionic slot system shown to players.

Blocking issue: #16920 (@Coolthulhu, please add to OP).

(I've posted a markdown version of the list presented in that issue's OP, see [last comment ATM](https://github.com/CleverRaven/Cataclysm-DDA/issues/16920#issuecomment-274269139).)

> The major concerns would be: Does it have bugs" Will the community take it well? Is it balanced enough?

Same answer for all 3 questions.

There is a deadlock: the system is not enabled by default, so no one gets to test it, so no one cares about "balancing" it, so it doesn't get enabled.

Balancing depends on estabilishing what a "bionic slot" represents, as discussed in #16920 (towards the end ATM). Percentage of volume? Absolute volume? Abstract magic number?

Do direct all balancing discussion there.

-----

> Hiding the system if it cannot be implemented in time would likely be the best solution.

> If anything, the system should not have been implemented in the first place if it was not functional as-is.

It seemed "functional" to me when testing.

-----

Hiding slot display could be based on checking `DEBUG_CBM_SLOTS` (relevant: PR #16915).

**Comment by DangerNoodle** [comment]
> It seemed "functional" to me when testing.

Functional as in it actually had any effect when you went over the CBM limit? If not, then it absolutely was not functional when first implemented.

**Comment by keyspace** [comment]
> Functional as in it actually had any effect when you went over the CBM limit?

For a character with `DEBUG_CBM_SLOTS`, going over the slot limit is not possible.

**Comment by DangerNoodle** [comment]
I see. Peculiar way to implement it.

**Comment by keyspace** [comment]
Would be nice to mention @illi-kun IMO.

**Comment by Coolthulhu** [comment]
> There is a deadlock: the system is not enabled by default, so no one gets to test it, so no one cares about "balancing" it, so it doesn't get enabled.

It was "enabled" for a while. As in, shown to players, who then talked about how to balance it.
Nothing came out of it, despite many months of possible feedback.

So for now it's better to just hide the system - that does nothing - and get back to it once the stable is out.

I'll try to implement bionic limits in my mods first. Unless someone beats me to it in the core.

**Comment by illi-kun** [comment]
@keyspace you're right about #16920. I have some propositions in the referenced issue not answered for months, and I wasn't ready to implement a big portion of work just to get another "I disagree" answer from @Coolthulhu on the latest review stage, like it happens for other parts of bionics slots system implementation.

So the smoothest way to get the system implemented is to wait Coolthulhu's mod.

**Comment by Coolthulhu** [comment]
> I wasn't ready to implement a big portion of work just to get another "I disagree" answer from @Coolthulhu on the latest review stage, like it happens for other parts of bionics slots system implementation.

One of my jobs here is to prevent half-baked systems from ruining the game for a lot of people. If anything, I'm not strict enough at the job - my biggest fuckups in the development invariably involved me loosening the standards from "it needs to be an improvement" to "improvements will follow up soon" after being asked nicely.

If you expect trouble or aren't sure if you know what to do, propose changes before implementing them. I won't take the blame for giving the (negative) answer too late when you were the one to ask the question too late.

**Comment by DangerNoodle** [comment]
I am afraid I must agree. I feel as though many features as of late should not have been added to the game due to not being complete in initial implementation. This is poor code quality because it relies on the assumption that it will become function in the future, which is not guaranteed.

Already I am concerned that https://github.com/CleverRaven/Cataclysm-DDA/issues/20358 will present another example of an incomplete implementation that might not be finished before the next stable build. However, unlike the vitamin system, this is a minor consistency problem, not a major disruption.

**Comment by illi-kun** [comment]
> If you expect trouble or aren't sure if you know what to do, propose changes before implementing them.

That's what I did in #16920 and got no response.

**Comment by DangerNoodle** [comment]
I see a long page of discussion on this thread, and I myself gave what I thought was a sensible suggestion to reduce the slot size of faulty bionics for balancing reasons (they are annoying enough already, a token slot size is nothing more than a nod to realism, not a valid gameplay factor).

**Comment by Coolthulhu** [comment]
> That's what I did in #16920 and got no response.

Last of your posts there doesn't really propose much, just changing units. Hard to say much there.
I'll try to write something regarding the new units, but without a balance-related idea, it doesn't lead anywhere yet.

**Comment by Parahacker** [comment]
To me, the CBM restriction per body part system is a prime candidate for being a mod, not a core feature - even when it's all perfectly balanced, I can see wanting to turn it off and go full cyber monster on a game.

As it stands, it's severely limiting considering that a)uninstalled cbms turn into burnt out bionics b)most of the niftiest cbms are uncraftable; install a valuable mod like the probability travel cbm, you're stuck with it even if most of the time you don't use it (and the other times you *really* use it). If I could 'swap out' bionics (with the normal risks of self-surgery, of course) then it'd be much more workable; bump up the body points about 10 or so per body part, and it would be a comfortable level of trade-offs. But the bionics I consider indispensable already overfill the current limits, much less the nice-but-optional ones like the UPS or more power sources.

**Comment by kevingranade** [comment]
> even when it's all perfectly balanced, I can see wanting to turn it off and go full cyber monster on a game.

This isn't a reason to have it be mod only, it should be the default, and if someone wants to disable it in a mod they can.  All you have to do is set CBM sizes to 0.
